pkg/entrypoint/server_test.go

```diff
@@ -0,0 +1,197 @@
+		testcase := testCase{
+			input: []Job{
+				{
+					Name:    "test1",
+					Command: "sleep",
+					Args:    []string{"1"},
+				},
+			},
+			expected: []statusResponse{
+				{Jobs: []job{{Name: "test1", Status: "Running"}}},
+				{Jobs: []job{{Name: "test1", Status: "Completed"}}},
+			},
+		}
+
+		cancel := startRunner(apiAddr, testcase.input)
+		defer cancel()
+
+		for _, expected := range testcase.expected {
+			Eventually(gotStatus, 10, 0.5).Should(Equal(&expected))
+		}
+	})
+
+	It("should state of command with exit code 1 become failed", func() {
+		testcase := testCase{
+			input: []Job{
+				{
+					Name:    "test1",
+					Command: "false",
+					Args:    []string{},
+				}},
+			expected: []statusResponse{
+				{Jobs: []job{{Name: "test1", Status: "Failed"}}},
+			},
+		}
+
+		cancel := startRunner(apiAddr, testcase.input)
+		defer cancel()
+
+		for _, expected := range testcase.expected {
+			Eventually(gotStatus, 10, 0.5).Should(Equal(&expected))
+		}
+	})
+
+	It("should state of unknown command become failed", func() {
+		testcase := testCase{
+			input: []Job{
+				{
+					Name:    "test1",
+					Command: "unknowncommand",
+					Args:    []string{},
+				}},
+			expected: []statusResponse{
+				{Jobs: []job{{Name: "test1", Status: "Failed"}}},
+			},
+		}
+
+		cancel := startRunner(apiAddr, testcase.input)
+		defer cancel()
+
+		for _, expected := range testcase.expected {
+			Eventually(gotStatus, 10, 0.5).Should(Equal(&expected))
+		}
+	})
+
+	It("should state of not permission command become failed", func() {
+		testcase := testCase{
+			input: []Job{
+				{
+					Name:    "test1",
+					Command: "./testresources/testscript.sh",
+					Args:    []string{},
+				}},
+			expected: []statusResponse{
+				{Jobs: []job{{Name: "test1", Status: "Failed"}}},
+			},
+		}
+
+		cancel := startRunner(apiAddr, testcase.input)
+		defer cancel()
+
+		for _, expected := range testcase.expected {
+			Eventually(gotStatus, 10, 0.5).Should(Equal(&expected))
+		}
+	})
+
+	It("should two sucessful command is executed in series", func() {
+
+		testcase := testCase{
+			input: []Job{
+				{
+					Name:    "test1",
+					Command: "echo",
+					Args:    []string{"1"},
+				},
+				{
+					Name:    "test2",
+					Command: "sleep",
+					Args:    []string{"1"},
+				}},
+			expected: []statusResponse{
+				{Jobs: []job{{Name: "test1", Status: "Completed"}, {Name: "test2", Status: "Running"}}},
+				{Jobs: []job{{Name: "test1", Status: "Completed"}, {Name: "test2", Status: "Completed"}}},
+			},
+		}
+
+		cancel := startRunner(apiAddr, testcase.input)
+		defer cancel()
+
+		for _, expected := range testcase.expected {
+			Eventually(gotStatus, 10, 0.5).Should(Equal(&expected))
+		}
+	})
+	It("should second command is pended when first command fails", func() {
+
+		testcase := testCase{
+			input: []Job{
+				{
+					Name:    "test1",
+					Command: "false",
+					Args:    []string{""},
```

Comment:
```suggestion
					Args:    []string{},
```

Of course, it does not affect the behavior of the command.